Automattic/wpcom-proxy-request
# Patch File objects sent as form data to work around a bug in Chrome

Before sending request data over `window.postMessage` to the proxy iframe, we find all `File` instances in the form data and create a new `File` instance based on the existing `File`. That forces the `File` object's storage to be a `Blob` instead of being backed by a file on disk. That works around a bug in Chrome where `File` instances with `has_backing_file` flag cannot be sent over a process boundary when site isolation is on.

**Note:** I'm not attempting to detect Chrome and just patch the `File`s for all. I tried to detect Chrome by looking at `window.chrome.webstore`, but that property isn't always there.

This PR also removes an old workaround for Firefox where the browser couldn't send a `File` over `window.postMessage` at all. That bug has been fixed for several years now.

**See also:**
https://bugs.chromium.org/p/chromium/issues/detail?id=866805
https://bugs.chromium.org/p/chromium/issues/detail?id=631877
https://github.com/Automattic/wp-calypso/pull/26604

**Testing instructions:**
- build with `make build`
- copy `build/index.js` to your Calypso repo: `node_modules/wpcom-proxy-request/build/index.js`
- when editing a post in Calypso, try to upload a featured image with drag & drop. Verify in Network Monitor that a POST request was sent to `/sites/$site/media/new`, that it had the whole file in the body and that it returned a success response.
- test in Chrome to verify that the bug is fixed, in other browsers to verify they didn't break.
